### PR TITLE
Km lucas

### PR DESCRIPTION
**Pull request by pclucas14**
0. move the train/dev/test splitting into `maybe_filter_hf_dataset_by_task`
1. allow multiple labels in data loader to compute Rouge
2. added `dataloader_num_workers` arg

3. training QA and eval QA scripts
4. scripts to create RAG datasets
5. changed a few things in the NQAcallback

### CODE REVIEW

File: mttl/models/containers/selectors/base.py [hunk 1]
```diff
@@ -539,11 +539,17 @@ def get_merging_weights(self, **selector_kwargs) -> Dict:
 @dataclass
 class KnowledgeExtractorSelectorConfig(SelectorConfig):
     field_name: str = "task_names"
-    ke_experts_prefix: str = "KE"
+    ke_expert_name: str = None
 
 
 @Selector.register("ke_selector", KnowledgeExtractorSelectorConfig)
```
Comment (sordonia) [inline mttl/models/containers/selectors/base.py:545]: can you put this into another selector? e.g. selectors/knowledge_modules.py

File: mttl/models/hf/trainer.py [hunk 2]
```diff
@@ -138,15 +149,23 @@ def _save(self, output_dir: Optional[str] = None, state_dict=None):
 class LMTrainer(ExpertModelTrainer):
     """Standard next-token prediction objective"""
 
-    def compute_loss(self, model, inputs, return_outputs=False):
-        input_ids = inputs["input_ids"]
-        labels = inputs["labels"]
-        attention_mask = inputs["attention_mask"]
+    # @InfoContainer.wrap_with_context(idx=1)
```
Comment (sordonia) [inline mttl/models/hf/trainer.py:152]: remove comment

File: projects/kms/utils/concatenate_datasets.py [hunk 3]
```diff
@@ -48,6 +88,7 @@ def main():
         print(f"Length of concatenated dataset: {len(concatenated_splits[split])}")
 
     # Save concatenated dataset
+    breakpoint()
```
Comment (sordonia) [inline projects/kms/utils/concatenate_datasets.py:91]: remove breakpoint

**Review by sordonia — APPROVED** [review]

**Review by shuishen112 — APPROVED** [review]
